src/components/comment-box.tsx

```diff
@@ -1,18 +1,60 @@
 import * as React from 'react';
+import { useMutation } from 'react-apollo';
+import Router from 'next/router';
+import Picker from 'react-giphy-picker';
+
+import { NewsItemModel } from '../data/models';
+import { POST_COMMENT_MUTATION } from '../data/mutations/post-comment-mutation';
+
+export interface ICommentsBoxProps {
+  newsItem: NewsItemModel;
+}
+
+export function CommentBox(props: ICommentsBoxProps): JSX.Element {
+  const { newsItem } = props;
+
+  const [text, setText] = React.useState<string>('');
+  const [isGif, setIsGif] = React.useState<boolean>(false);
+
+  const [postComment] = useMutation(POST_COMMENT_MUTATION, {
+    onError(error) {
+      Router.push('/login');
+    },
+    variables: { id: 999, parent: newsItem.id, text, isGif },
+  });
 
-export function CommentBox(): JSX.Element {
   return (
     <tr>
       <td colSpan={2} />
       <td>
-        <form method="post" action="comment">
+        <form
+          method="post"
+          action="comment"
+          onSubmit={(e) => {
+            e.preventDefault();
+            text.trim() !== '' ? postComment() : alert('Cannot post an empty comment!');
+            return Router.reload();
+          }}
+        >
```

Comment:
I try to handle some cases like User posting empty comment or user hasn't logged in yet